src/Judge/NTS.Judge/Core/Behinds/Adapters/RanklistBehind.cs

```diff
@@ -50,6 +52,15 @@ protected override async Task<bool> PerformInitialization(params IEnumerable<obj
             return false;
         }
         Ranklist = new Ranklist(ranking);
+        var enduranceEvent = await _events.Read(0);
+        if (enduranceEvent == null)
+        {
+            NotifyHelper.Warn("Event is not started yet");
+            return false;
+        }
+        var officials = await _officials.ReadAll();
+        var ranklistDocument = new RanklistDocument(Ranklist, enduranceEvent, officials);
+        Document = ranklistDocument;
```

Comment:
I am thinking that we don't need to duplicate this here and in the Select method below. Can we instead expose CreateDocument method which will return the document from the currently selected Ranklist, for example